sdk/ai/azure-ai-resources/azure/ai/resources/index/_dataindex/entities/_builders/data_index_func.py

```diff
@@ -244,23 +244,30 @@ def data_index_incremental_update_acs(
     from azure.ai.resources.index._dataindex.dsl._pipeline_decorator import pipeline
 
     crack_and_chunk_and_embed_component = get_component_obj(ml_client, LLMRAGComponentUri.LLM_RAG_CRACK_AND_CHUNK_AND_EMBED)
-    update_acs_index_component = get_component_obj(ml_client, LLMRAGComponentUri.LLM_RAG_UPDATE_ACS_INDEX)
+
+    if data_index.index.type == DataIndexTypes.ACS:
+        update_index_component = get_component_obj(ml_client, LLMRAGComponentUri.LLM_RAG_UPDATE_ACS_INDEX)
+    elif data_index.index.type == DataIndexTypes.PINECONE:
+        update_index_component = get_component_obj(ml_client, LLMRAGComponentUri.LLM_RAG_UPDATE_PINECONE_INDEX)
+    else:
```

Comment:
For my learning purpose, it seems we don't support FAISS index update, correct?

Reply:
We do, look for `data_index_faiss` as we handle FAISS differently. I updated this method to work for all hosted indexes (which at this point, are only ACS and Pinecone) hence why we only accept ACS and Pinecone here.